crates/wasmtime/src/config.rs

```diff
@@ -1600,7 +1605,23 @@ impl Config {
                 .compiler_config
                 .ensure_setting_unset_or_given("unwind_info", "true")
             {
-                bail!("compiler option 'unwind_info' must be enabled profiling");
+                bail!("compiler option 'unwind_info' must be enabled for profiling");
+            }
+        }
+
+        if !self.native_unwind_info
+            && target.operating_system != target_lexicon::OperatingSystem::Windows
+        {
+            if self.profiling_strategy == ProfilingStrategy::None {
+                if let Strategy::Cranelift = self.compiler_config.strategy {
+                    unsafe {
+                        self.cranelift_flag_set("unwind_info", "false");
+                    }
+                }
+            } else {
+                bail!(
+                    "compiler option 'unwind_info' can not be disabled when profiling is enabled"
+                );
```

Comment:
Yeah, I think it would be slightly nicer to do something like

```rust
if self.profiling_strategy != ProfilingStrategy::None ||
    target == target_lexicon::OperatingSystem::Windows
{
    self.cranelift_flag_set("unwind_info", "true");
}

if !self.compiler_config.ensure_setting_unset_or_given(
    "unwind_info", 
    self.native_unwind_info.unwrap_or(DEFAULT_UNWIND_INFO),
) {
    bail!("`native_unwind_info` cannot be disabled when profiling or on Windows");
}
```

Reply:
- If profiling enabled, always include unwind info, and ignore `self.native_unwind_info`, even if set by user. We likely want to add this to the functoin comments?
- I assume `DEFAULT_UNWIND_INFO` to true?